### ISSUE TITLE
Build error with PGI on Titan

### ISSUE THREAD

**Issue by jychoi-hpc** [ornladios/ADIOS2#1287]
While trying to build the recent version of Adios2 on Titan with PGI, I got the following error:
```
"/lustre/atlas/world-shared/csc143/jyc/titan/src/adios2-devel/thirdparty/nlohma
          nn_json/nlohmann_json/src/single_include/nlohmann/json.hpp", line
          7132: internal error: assertion failed at: "../src/mem_manage.c",
          line 1540

                  auto last_token = get_token_string();
                       ^

1 catastrophic error detected in the compilation of "/lustre/atlas/world-shared/csc143/jyc/titan/src/adios2-devel/source/adios2/toolkit/format/dataman/DataManSerializer.cpp".
Compilation aborted.
pgc++-Fatal-/opt/pgi/18.4.0/linux86-64/18.4/bin/pggpp1 TERMINATED by signal 6
Arguments to /opt/pgi/18.4.0/linux86-64/18.4/bin/pggpp1
/opt/pgi/18.4.0/linux86-64/18.4/bin/pggpp1 --llalign -Dunix -D__unix -D__unix__ -Dlinux -D__linux -D__linux__ -D__NO_MATH_INLINES -D__LP64__ -D__x86_64 -D__x86_64__ -D__LONG_MAX__=9223372036854775807L '-D__SIZE_TYPE__=unsigned long int' '-D__PTRDIFF_TYPE__=long int' -D__extension__= -D__amd_64__amd64__ -D__k8 -D__k8__ -D__SSE__ -D__MMX__ -D__SSE2__ -D__SSE3__ -D__SSSE3__ -D__SSE4A__ -D__ABM__ -D__PGI -D_GNU_SOURCE -D_PGCG_SOURCE -I/lustre/atlas/world-shared/csc143/jyc/titan/src/adios2-devel/source -I/lustre/atlas/world-shared/csc143/jyc/titan/src/adios2-devel/build/source -I/lustre/atlas/world-shared/csc143/jyc/titan/src/adios2-devel/bindings/C -I/lustre/atlas/world-shared/csc143/jyc/titan/src/adios2-devel/bindings/CXX11 -I/lustre/atlas/world-shared/csc143/jyc/titan/src/adios2-devel/build/thirdparty/KWSys/adios2sys -I/lustre/atlas/world-shared/csc143/jyc/titan/src/adios2-devel/thirdparty/pugixml/pugixml/src -I/lustre/atlas/world-shared/csc143/jyc/titan/src/adios2-devel/thirdparty/nlohmann_json/nlohmann_json/src/single_include -I/opt/cray/mpt/7.6.3/gni/mpich-pgi/15.3/include -I/opt/cray/libsci/16.11.1/PGI/15.3/x86_64/include -I/opt/nvidia/cudatoolkit9.1/9.1.85_3.10-1.0502.df1cc54.3.1/include -I/opt/nvidia/cudatoolkit9.1/9.1.85_3.10-1.0502.df1cc54.3.1/extras/CUPTI/include -I/opt/nvidia/cudatoolkit9.1/9.1.85_3.10-1.0502.df1cc54.3.1/nvvm/include -I/opt/cray/rca/1.0.0-2.0502.60530.1.63.gem/include -I/opt/cray/alps/5.2.4-2.0502.9774.31.12.gem/include -I/opt/cray/xpmem/0.1-2.0502.64982.5.3.gem/include -I/opt/cray/gni-headers/4.0-1.0502.10859.7.8.gem/include -I/opt/cray/pmi/5.0.12/include -I/opt/cray/ugni/6.0-1.0502.10863.8.28.gem/include -I/opt/cray/udreg/2.3.2-1.0502.10518.2.17.gem/include -I/opt/cray/wlm_detect/1.0-1.0502.64649.2.2.gem/include -I/opt/cray/krca/1.0.0-2.0502.67049.8.22.gem/include -I/opt/cray-hss-devel/7.2.0/include -I- -I/opt/pgi/18.4.0/linux86-64/18.4/include-gcc60 -I/opt/pgi/18.4.0/linux86-64/18.4/include -I/opt/gcc/6.3.0/snos/include/g++ -I/opt/gcc/6.3.0/snos/include/g++/x86_64-suse-linux -I/opt/gcc/6.3.0/snos/include/g++/backward -I/opt/gcc/6.3.0/snos/lib/gcc/x86_64-suse-linux/6.3.0/include -I/usr/local/include -I/opt/gcc/6.3.0/snos/include -I/opt/gcc/6.3.0/snos/lib/gcc/x86_64-suse-linux/6.3.0/include-fixed -I/usr/include -D__CRAYXE -D__CRAY_INTERLAGOS -D__CRAYXT_COMPUTE_LINUX_TARGET -DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX --preinclude _cplus_preinclude.h --preinclude_macros _cplus_macros.h --gnu_version=60300 -D__pgnu_vsn=60300 -g --dwarf2 --c++11 --gnu_extensions -q -o /tmp/pgc++8bMm0hSXHPxs.il /lustre/atlas/world-shared/csc143/jyc/titan/src/adios2-devel/source/adios2/toolkit/format/dataman/DataManSerializer.cpp
make[2]: *** [source/adios2/CMakeFiles/adios2.dir/toolkit/format/dataman/DataManSerializer.cpp.o] Error 127
make[2]: *** Waiting for unfinished jobs....
```

I was able to build before on Titan with PGI by following instructions on the document. But, I am having this error with the recent version. I am wondering if there is any fix.

**Comment by chuckatkins** [comment]
Do you have 18.10 available?  We run 18.10 I. Our CI so it's a known good version.

**Comment by jychoi-hpc** [comment]
PGI 18.10 works on Titan. Thank you for the quick information.
